app/logic/collector_logic.rb

```diff
@@ -23,6 +24,10 @@ def ping_times_guard
         return Pipeline.new(400, p[:payload], 'Wrong payload_version')
       end
 
+      REQUIRED_PAYLOAD_FIELDS.each do |field|
+        return Pipeline.new(400, p[:payload], "No payload field: #{field}") unless collector.payload =~ /#{field}/
+      end
```

Comment:
It generally works, but we should check the same for every record in the pipeline (now it's enough that one of them has required fields).